torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -898,6 +927,49 @@ def _run_param_init_fn():
         # For validating execution order across ranks
         self._exec_order_data = _ExecOrderData()
 
+    def _init_param_exec_order_wrap_policy(
+        self,
+        module: nn.Module,
+        process_group: Optional[ProcessGroup] = None,
+        sharding_strategy: Optional[ShardingStrategy] = None,
+        cpu_offload: Optional[CPUOffload] = None,
+        auto_wrap_policy: Optional[Callable] = None,
+        backward_prefetch: Optional[BackwardPrefetch] = None,
+        mixed_precision: Optional[MixedPrecision] = None,
+        ignored_modules: Optional[Iterable[torch.nn.Module]] = None,
+        param_init_fn: Optional[Callable[[nn.Module], None]] = None,
+        device_id: Optional[Union[int, torch.device]] = None,
+        sync_module_states: bool = False,
+    ) -> None:
+        # The initial FSDP wrapping is done with auto_wrap_policy.init_policy
+        self.__init__(
+            module=module,
+            process_group=process_group,
+            sharding_strategy=sharding_strategy,
+            cpu_offload=cpu_offload,
+            auto_wrap_policy=auto_wrap_policy.init_policy,
+            backward_prefetch=backward_prefetch,
+            mixed_precision=mixed_precision,
+            ignored_modules=ignored_modules,
+            param_init_fn=param_init_fn,
+            device_id=device_id,
+            sync_module_states=sync_module_states,
+        )
+        self._param_exec_order_policy : bool = True
```

Comment:
nit: I think the syntax we have been going with is no space before the colon.
```suggestion
        self._param_exec_order_policy: bool = True
```
https://peps.python.org/pep-0526/